src/main/java/net/dv8tion/jda/api/entities/guild/SystemChannelFlag.java

```diff
@@ -0,0 +1,138 @@
+
+import javax.annotation.Nonnull;
+import java.util.EnumSet;
+import java.util.HashMap;
+
+/**
+ * Represents Discord's system channel flags of a guild. A system channel is a
+ * unique channel in a guild where guild notices such as welcome messages and boost events
+ * are posted. System channel flags configure suppressions of these messages and events. According
+ * to the <a href="https://discord.com/developers/docs/">Discord API docs<a/>, system channel flags
+ * are represented as one-hot encoded bitmasks, where each bit in a binary string represents whether
+ * a particular flag is enabled. This class contains methods to convert these bitmasks to an {@link EnumSet}
+ * of this class, which is the representation of system channel flags used by users of JDA.
+ *
+ * @see <a href="https://discord.com/developers/docs/resources/guild#guild-object-system-channel-flags">
+ *     System Channel Flags API documentation
+ *     <a/>
+ * */
+public enum SystemChannelFlag
+{
+
+    /**
+     * Suppress member join notifications.
+     * */
+    SUPPRESS_JOIN_NOTIFICATIONS(0),
+
+    /**
+     * Suppress server boost notifications.
+     * Identical to the "SUPPRESS_PREMIUM_SUBSCRIPTIONS" flag in the API
+     * docs.
+     * */
+    SUPPRESS_BOOST_NOTIFICATIONS(1),
+
+    /**
+     * Suppress server setup tips.
+     * Identical to the "SUPPRESS_GUILD_REMINDER_NOTIFICATIONS" flag in the
+     * API docs.
+     * */
+    SUPPRESS_SETUP_TIPS(2),
+
+    /**
+     * Hide member join sticker reply buttons.
+     * Identical to "SUPPRESS_JOIN_NOTIFICATION_REPLIES" flag in the
+     * API docs.
+     * */
+    HIDE_JOIN_REPLY_STICKERS(3),
+
+    /**
+     * Suppress role subscription purchase and renewal notifications.
+     * Identical to "SUPPRESS_ROLE_SUBSCRIPTION_PURCHASE_NOTIFICATIONS" flag
+     * in the API docs.
+     * */
+    SUPPRESS_ROLE_PURCHASE_NOTIFICATIONS(4),
+
+    /**
+     * Hide role subscription sticker reply buttons. Identical to
+     * "SUPPRESS_ROLE_SUBSCRIPTION_PURCHASE_NOTIFICATION_REPLIES" flag
+     * in the API docs.
+     * */
+    HIDE_ROLE_SUBSCRIPTION_STICKERS(5);
+
+    private final int offset;
+
+    private static final HashMap<Integer, SystemChannelFlag> OFFSET_TO_FLAG = new HashMap<>();
+
+    static
+    {
+        for (SystemChannelFlag flag : SystemChannelFlag.values())
+        {
+            OFFSET_TO_FLAG.put(flag.offset, flag);
+        }
+    }
+
+    SystemChannelFlag(int offset)
+    {
+        this.offset = offset;
+    }
+
+    /**
+     * Converts a bitmask representation of system channel flags into its
+     * {@link EnumSet}.
+     * @param bitmask the raw bitmask representing the system channel flags.
+     * @return an {@link EnumSet} of system channel flags represented by the input bitmask.
+     * @throws IllegalArgumentException if the input bitmask is invalid.
+     * */
+    @Nonnull
+    public static EnumSet<SystemChannelFlag> fromBitmask(final int bitmask)
```

Comment:
```suggestion
    public static EnumSet<SystemChannelFlag> getFlags(int raw)
```

Please refer to:
https://github.com/discord-jda/JDA/blob/e2c3b7439c46b23ca77fae804f75153d315d4b31/src/main/java/net/dv8tion/jda/api/requests/GatewayIntent.java#L265-L274